src/main/java/com/google/sps/servlets/UpdateVendorServlet.java

```diff
@@ -0,0 +1,122 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.appengine.api.blobstore.BlobKey;
+import com.google.appengine.api.blobstore.BlobstoreService;
+import com.google.appengine.api.blobstore.BlobstoreServiceFactory;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.EmbeddedEntity;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.Key;
+import com.google.appengine.api.datastore.KeyFactory;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.sps.data.HttpServletUtils;
+import com.google.sps.data.Vendor;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/**
+ * When the user submits the form, Blobstore processes the file upload and then forwards the request
+ * to this servlet to update a vendor information
+ */
+@WebServlet("/update-vendor")
+public class UpdateVendorServlet extends HttpServlet {
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    UserService userService = UserServiceFactory.getUserService();
+    
+    if (userService.isUserLoggedIn()) {
+      String firstName = HttpServletUtils.getParameter(request, "firstName", "");
+      String lastName = HttpServletUtils.getParameter(request, "lastName", "");
+      String phoneNumber = HttpServletUtils.getParameter(request, "phoneNumber", "");
+      
+      // Get values to add/update vendor's profile picture
+      String currentBlobKey = HttpServletUtils.getParameter(request, "blobKey", "");
+      BlobKey imageBlobKey = HttpServletUtils.getUploadedFileBlobKey(request, "imageFile");
+      String altText = HttpServletUtils.getParameter(request, "altText", "");
+      // If nothing was uploaded and there is a current picture, keep current
+      if (!currentBlobKey.isEmpty() && imageBlobKey == null) {
+        imageBlobKey = new BlobKey(currentBlobKey);
+      }
+
+      // Check values are not empty and valid
```

Comment:
This is very helpful, I think this should be a separate function that we can call from different servlets.

Reply:
+1

Reply:
Hey, just to be sure, you mean the check for values and the deletion from blobstore?